src/Client-SDK/Temporal.WorkflowClient/ClientApi.cs

```diff
@@ -0,0 +1,466 @@
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg>(string taskQueueMoniker, TWfArg workflowArgs, string signalName, CancellationToken cancelToken) where TWfArg : IDataValue { return null; }
+
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg, TSigArg>(string taskQueueMoniker, TWfArg workflowArgs, string signalName, TSigArg signalArgs) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg, TSigArg>(string taskQueueMoniker, TWfArg workflowArgs, string signalName, TSigArg signalArgs, CancellationToken cancelToken) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+
+        public Task<WorkflowRun> StartNewRunWithSignalAsync(IWorkflowExecutionConfiguration workflowRunConfig, string signalName) { return null; }
+        public Task<WorkflowRun> StartNewRunWithSignalAsync(IWorkflowExecutionConfiguration workflowRunConfig, string signalName, CancellationToken cancelToken) { return null; }
+
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, string signalName, TSigArg signalArgs) where TSigArg : IDataValue { return null; }
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, string signalName, TSigArg signalArgs, CancellationToken cancelToken) where TSigArg : IDataValue { return null; }
+
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName) where TWfArg : IDataValue { return null; }
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName, CancellationToken cancelToken) where TWfArg : IDataValue { return null; }
+
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg, TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName, TSigArg signalArgs) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+        public Task<WorkflowRun> StartNewRunWithSignalAsync<TWfArg, TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName, TSigArg signalArgs, CancellationToken cancelToken) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+
+        public Task<WorkflowRun> StartNewRunWithSignalAsync(IWorkflowExecutionConfiguration workflowRunConfig, PayloadsCollection workflowArgs, string signalName, PayloadsCollection signalArgs, CancellationToken cancelToken) { return null; }
+        #endregion StartNewRunWithSignalAsync(..)
+
+
+        #region TryStartNewRunWithSignalAsync(..)
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync(string taskQueueMoniker, string signalName) { return null; }
```

Comment:
Implementing the relatively common Try-patern on .NET for async methods.

Reply:
What happens when you throw inside an async function? Can we just leave it at that instead of all of these `Try` overloads (even if it is a common pattern)?